decidim-proposals/app/commands/decidim/proposals/admin/answer_proposal.rb

```diff
@@ -46,12 +46,12 @@ def answer_proposal
             attributes = {
               state: form.state,
               answer: form.answer,
-              answered_at: Time.current,
               cost: form.cost,
               cost_report: form.cost_report,
               execution_period: form.execution_period
             }
 
+            attributes[:answered_at] = form.state != "not_answered" ? Time.current : nil
             attributes[:state_published_at] = Time.current if !initial_has_state_published && form.publish_answer?
```

Comment:
I think we also have to set this back to `nil` for the "Not answered" filter to work in the front-end:
![Not answered filter](https://user-images.githubusercontent.com/864340/162176467-5023a51c-c0ed-4cb5-a224-8293f344c010.png)

The "Not answered" is using the `state_not_published` filter as defined here:
https://github.com/decidim/decidim/blob/c93dae384196adc86eb0683257c2b59f2b4295bb/decidim-proposals/app/helpers/decidim/proposals/proposals_helper.rb#L44

And this is matching all records that have the `state_published_at` set to `nil`:
https://github.com/decidim/decidim/blob/c93dae384196adc86eb0683257c2b59f2b4295bb/decidim-proposals/app/models/decidim/proposals/proposal.rb#L61

Sorry I did not notice this before because I just assumed the "Not answered" filter was calling the `not_answered` scope...

Reply:
No problem, I've changed it and also added in the specs.